geth/common/types.go

```diff
@@ -206,7 +204,7 @@ type TxQueueManager interface {
 	QueueTransaction(tx *QueuedTx) error
 
 	// WaitForTransactions blocks until transaction is completed, discarded or timed out.
-	WaitForTransaction(tx *QueuedTx) error
+	WaitForTransaction(tx *QueuedTx) RawCompleteTransactionResult
```

Comment:
Doesn't `Result` part assume that transaction is completed? I'd renamed it to `RawTransactionResult` if that makes sense.